src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java

```diff
@@ -154,26 +150,7 @@ private void addLinkedFileFromURL(BibDatabaseContext databaseContext, URL url, B
                     preferences.getFilePreferences(),
                     ExternalFileTypes.getInstance());
 
-            try {
-                URLDownload urlDownload = new URLDownload(newLinkedFile.getLink());
-                BackgroundTask<Path> downloadTask = onlineFile.prepareDownloadTask(targetDirectory, urlDownload);
-                downloadTask.onSuccess(destination -> {
-                    LinkedFile downloadedFile = LinkedFilesEditorViewModel.fromFile(
-                            destination,
-                            databaseContext.getFileDirectories(preferences.getFilePreferences()),
-                            ExternalFileTypes.getInstance());
-                    entry.addFile(downloadedFile);
-                    dialogService.notify(Localization.lang("Finished downloading full text document for entry %0.",
-                            entry.getCitationKey().orElse(Localization.lang("undefined"))));
-                });
-                downloadTask.titleProperty().set(Localization.lang("Downloading"));
-                downloadTask.messageProperty().set(
-                        Localization.lang("Fulltext for") + ": " + entry.getCitationKey().orElse(Localization.lang("New entry")));
-                downloadTask.showToUser(true);
-                Globals.TASK_EXECUTOR.execute(downloadTask);
-            } catch (MalformedURLException exception) {
-                dialogService.showErrorDialogAndWait(Localization.lang("Invalid URL"), exception);
-            }
+            onlineFile.download();
```

Comment:
I think with this change, downloading a pdf doesn't show any progress any more since the `onlineFile` is not added to the entry until the download succeeds.

Reply:
Not exactly sure, at this point we add the online fie link and then call the download method which also shows the progress
https://github.com/JabRef/jabref/blob/c8f7be89ad704065474decb61851e42416caf88d/src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java#L415-L454